index.html

```diff
@@ -3102,21 +3122,48 @@ <h3>Verify Proof Sets and Chains</h3>
 Set <var>allProofs</var> to </var><var>securedDocument</var>.<var>proof</var>.
           </li>
           <li>
-For each <var>proof</var> in <var>allProofs</var>, run steps 2 through 11 of the algorithm in
-section <a href="#verify-proof"></a>; if no exceptions are raised, associate
-the <var>isProofVerified</var> value with this <var>proof</var>.
+For each <var>proof</var> in <var>allProofs</var> do the following steps:
           </li>
           <li>
-For each <var>proof</var> in <var>allProofs</var> that contains a
-<code>previousProof</code> attribute, modify the associated
-<var>isProofVerified</var> by iteratively checking that the proof identified by
-the <code>id</code> value in <var>previousProof</var>, along with every
-<code>previousProof</code> up the chain, is valid. If <code>previousProof</code> is an
-array, then all previous proofs identified by the array elements
-MUST be valid for the current proof to be considered valid. Cycle detection,
-such as keeping a list of previously visited proofs, MUST be used to guard
-against proof cycles. If a cycle is detected in a proof chain,
-then a `PROOF_CHAIN_CYCLE_ERROR` MUST be raised.
+Let <var>matchingProofs</var> be an empty set.
+          </li>
+          <li>
+If <var>proof</var> contains a <code>previousProof</code> attribute and that
+attribute is a string, add the element from <var>allProofs</var> with a matching
+<code>id</code> attribute to matchingProofs. If it does not exist, an error MUST
```

Comment:
Again, what is the "it" that might not exist?
```suggestion
<code>id</code> attribute to matchingProofs. If it does not exist, an error MUST
```